credential/concept/registry.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package concept
+
+import (
+	"github.com/nuts-foundation/go-did"
+)
+
+// Registry defines the interface for accessing loaded concepts and using the templates
+// to generate queries and transform results.
+type Registry interface {
+	// ConceptTemplates returns a mapping of concept names to parsed templates.
+	ConceptTemplates() map[string][]Template
+	// Load template files from disk and parse them.
+	LoadTemplates() error
+	// Adds a template from a string source
+	AddFromString(concept string) error
+	// QueryFor creates a query for the given concept.
+	// The query is preloaded with required fixed values like the type.
```

Comment:
'Loading' a data type (preloading) sounds a bit strange here
`The query contains fixed values required for the concept`

Reply:
The query params used are only valid within a certain concept. It's up to the DB query implementation to transform those to DB compatible queries. Therefor a query is created in the context of a concept.